test/models/test_mamba.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from tinygrad.helpers import CI
+from tinygrad import Device
+from examples.mamba import Mamba, generate
+from transformers import AutoTokenizer
+
+PROMPT = 'Why is gravity '
+TOKENIZER = AutoTokenizer.from_pretrained("EleutherAI/gpt-neox-20b")
+
+@unittest.skipIf(CI and Device.DEFAULT in ["CLANG"], "Inference hack broken on CLANG")
```

Comment:
What's broken on CLANG and why?

Reply:
After some digging, it's not what I thought.  
(logits[:, -1, :]).softmax().argmax(axis=-1).item() ALWAYS returns 50279 which is an empty string
This make me feel like softmax may be an issue cuz the probability of next token being empty should be the least, right?

Reply:
After some testing on a Tensor/torch.arange(16).reshape(4,4):
.softmax works properly
.argmax works properly
.softmax.argmax gets incorrect result ONLY FOR CLANG=1, works on CUDA=1

Reply:
That's not a good excuse to skip a test.

Reply:
Issue #3552